### PR TITLE
Add `func-style` ESLint rule

### PR DESCRIPTION
**Pull request by ehmicky**
This adds the [`func-style`](https://eslint.org/docs/rules/func-style) ESLint rule.

### CODE REVIEW

File: src/commands/functions/create.js [hunk 1]
```diff
@@ -152,40 +150,43 @@ async function pickTemplate() {
     },
   })
   return chosentemplate
-  function filterRegistry(registry, input) {
```
Comment (ehmicky) [inline src/commands/functions/create.js:155]: Those functions can be safely moved to the top-level scope.

**Review by erezrokah — CHANGES_REQUESTED** [review]
Should we enable https://eslint.org/docs/rules/func-style#allowarrowfunctions?

**Comment by ehmicky** [comment]
That works :+1: 

Should we try to ensure consistency (`const myFunc = function(...){...}` vs `const myFunc = (...) => {...}`). Both styles are almost the same and work, but would it be useful to enforce consistency over one specific style?
If so, which ESLint rule would be the best to enforce this? I have struggled with this specific linting problem in my personal projects, but there might be a solution.

**Comment by erezrokah** [comment]
> That option is always enabled when the rule is set to `expression` (the default),

Cool, didn't notice that.

I would also prefer to choose one of the two styles. Don't think it is a blocker for this PR (especially if we don't have built in support for it).

**Comment by ehmicky** [comment]
Rebased. Ready for another review :+1:

**Review by erezrokah — APPROVED** [review]

**Comment by ehmicky** [comment]
This is enforceable with `no-restricted-syntax`. However, enforcing arrow functions would also require writing:

```js
module.exports = {
  onBuild() {...}
}
```

As:

```js
module.exports = {
  onBuild: () => {...}
}
```

Which might be more verbose?

Also, this would not come with autofixing, so I am thinking this specific rule might be more of a nuisance than a help there, and leaving inconsistent style might be more practical. What do you think?

**Comment by erezrokah** [comment]
> Also, this would not come with autofixing, so I am thinking this specific rule might be more of a nuisance than a help there, and leaving inconsistent style might be more practical. What do you think?

Agreed

**Pull request merged**